packages/plugin-data/src/api/data/soql/queryFields.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ensureJsonArray, ensureJsonMap, ensureString, isJsonArray, toJsonMap } from '@salesforce/ts-types';
+import { Connection } from '@salesforce/core';
+import { Tooling } from '@salesforce/core/lib/connection';
+
+/**
+ * Class to represent a field when describing the fields that make up a query result
+ */
+export class Field {
+  public name: string;
+
+  public constructor(name: string) {
+    this.name = name;
+  }
+}
```

Comment:
Using a class for this doesn't feel quite right.  I think all of this will depend on what we want a public API to be.  Given this, I'd prefer to be conservative with what is exported so people don't try to use it directly.